### PR TITLE
NOISSUE - Support Cocos FileSystem

### PR DESCRIPTION
**Pull request by rodneyosodo**
This update introduces significant enhancements to support the Cocos AI system, focusing on documentation, dependency management, and conditional feature logic. Key changes allow for dynamic dataset paths and improved modularity in handling machine learning algorithms.

- **New Features**
	- Added support for the Cocos AI system with new documentation and features for secure workload execution.
	- Introduced conditional dataset loading and artifact management based on the Cocos feature across various algorithms.
	- Enhanced flexibility in handling dataset paths and metrics rendering for training processes.

- **Bug Fixes**
	- Improved error handling for dataset file access and parsing operations.

- **Documentation**
	- Added comprehensive documentation for the Cocos AI system including setup and operation instructions in `COCOS.md`.
	- Updated the README to reflect new Cocos integrations and examples.

- **Chores**
	- Updated dependencies and feature configurations across multiple projects to include local libraries and new functionalities.

- **Tests**
	- Add unit tests for the new functions in the lib module, including sending data via a Unix socket and saving results to a file.

### CODE REVIEW

File: burn-algorithms/addition-inference/Cargo.toml [hunk 1]
```diff
@@ -7,10 +7,10 @@ license.workspace = true
...
-input = []
+cocos = []
+read = []
 
 [dependencies]
 burn = { version = "0.13.2", default-features = false, features = ["ndarray"] }
 futures = "0.3.30"
-serde = { version = "1.0.203", features = ["derive"] }
-serde_json = "1.0.120"
+lib = { path = "../lib" }
```
Comment (WashingtonKK) [inline burn-algorithms/addition-inference/Cargo.toml:16]: Add new line.

File: burn-algorithms/addition/Cargo.toml [hunk 2]
```diff
@@ -11,9 +11,8 @@ publish = false
...
 wgpu = ["burn/wgpu"]
 cocos = []
-input = []
+read = []
 
 [dependencies]
 burn = { version = "0.13.2", features = ["ndarray", "wgpu"] }
-serde = { version = "1.0.203", features = ["derive"] }
-serde_json = "1.0.120"
+lib = { path = "../lib" }
```
Comment (WashingtonKK) [inline burn-algorithms/addition/Cargo.toml:18]: Add new line.

File: burn-algorithms/imdb/Cargo.toml [hunk 3]
```diff
@@ -20,3 +21,4 @@ csv = "1.3.0"
 derive-new = "0.6.0"
 serde = { version = "1.0.203", features = ["derive", "std"] }
 tokenizers = { version = "0.19.1", features = ["http", "onig"] }
+lib = { path = "../lib" }
```
Comment (WashingtonKK) [inline burn-algorithms/imdb/Cargo.toml:24]: Same here

File: burn-algorithms/mnist/Cargo.toml [hunk 4]
```diff
@@ -19,3 +20,4 @@ burn = { version = "0.13.2", features = [
 ] }
 burn-common = "0.13.2"
 serde = { version = "1.0.203", features = ["derive", "std"] }
+lib = { path = "../lib" }
```
Comment (WashingtonKK) [inline burn-algorithms/mnist/Cargo.toml:23]: New line

**Review by WashingtonKK — CHANGES_REQUESTED** [review]

**Review by SammyOina — APPROVED** [review]

**Pull request merged**